backend/dataall/modules/dataset_sharing/services/share_processors/lakeformation_process_share.py

```diff
@@ -64,10 +64,21 @@ def process_approved_shares(self) -> bool:
         if not self.shared_tables:
             log.info("No tables to share. Skipping...")
         else:
-            self.grant_pivot_role_all_database_permissions_to_source_database()
-            self.check_if_exists_and_create_shared_database_in_target()
-            self.grant_pivot_role_all_database_permissions_to_shared_database()
-            self.grant_principals_database_permissions_to_shared_database()
+            try:
+                if not self.check_catalog_account_exists_and_update_processor():
+                    success = False
+                    return success
```

Comment:
minor - let's just return false here

Reply:
It is minor, but this is still present in the last commited version

Reply:
ohh, I see. Thanks for the suggesion. I will update the PR accorindingly.  I was under the impression that `data_sharing_service.py` handles if I return false. but I see from the code it doesn't.